<documentation title="Flag returning high or infinite posts_per_page.">
    <standard>
    <![CDATA[
Using posts_per_page (or numberposts) with the value set to an unreasonably high number or setting nopaging to true opens up the potential for scaling issues if the query ends up querying thousands of posts.

You should always fetch the lowest number possible that still gives you the number of results you find acceptable.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: posts_per_page is not over limit (default 100).">
        <![CDATA[
$args = array(
    'posts_per_page' => <em>-1</em>,
    'posts_per_page' => <em>100</em>,
    'posts_per_page' => <em>'10'</em>,
);

$query_args['posts_per_page'] = <em>100</em>;
$query_args['posts_per_page'] = <em>'10'</em>;
        ]]>
        </code>
        <code title="Invalid: posts_per_page is over limit (default 100).">
        <![CDATA[
$args = array(
    'posts_per_page' => <em>999</em>,
);

_query_posts( 'nopaging=1&posts_per_page=999' );
        ]]>
        </code>
    </code_comparison>
    <code_comparison>
        <code title="Valid: numberposts is not over limit (default 100).">
        <![CDATA[
$args = array(
    'numberposts' => -1,
    'numberposts' => 100,
    'numberposts' => '10',
);

$query_args['numberposts'] = '-1';
        ]]>
        </code>
        <code title="Invalid: numberposts is over limit (default 100).">
        <![CDATA[
$args = array(
    'numberposts' => 999,
);

_query_posts( 'numberposts=999' );
        ]]>
        </code>
    </code_comparison>
</documentation>
